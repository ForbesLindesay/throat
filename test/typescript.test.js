const spawnSync = require('child_process').spawnSync;

test('typescript', () => {
  const result = spawnSync('npm', ['run', 'tsc']);
  expect(result.status).not.toBe(0);
  expect(result.stdout.toString('utf8').split('\n').filter(l => !/^\>/.test(l)).join('\n')).toMatchSnapshot();
});
