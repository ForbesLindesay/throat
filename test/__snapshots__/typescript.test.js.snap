// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript 1`] = `
"

test/typescript-example.ts(6,42): error TS2345: Argument of type '\\"foo\\"' is not assignable to parameter of type 'number'.
test/typescript-example.ts(7,7): error TS2322: Type 'Promise<number>' is not assignable to type 'Promise<string>'.
  Type 'number' is not assignable to type 'string'.
test/typescript-example.ts(12,42): error TS2345: Argument of type '\\"foo\\"' is not assignable to parameter of type 'number'.
test/typescript-example.ts(13,7): error TS2322: Type 'Promise<number>' is not assignable to type 'Promise<string>'.
test/typescript-example.ts(20,32): error TS2554: Expected 2 arguments, but got 1.
test/typescript-example.ts(21,33): error TS2554: Expected 2 arguments, but got 1.
test/typescript-example.ts(22,7): error TS2322: Type 'Promise<number>' is not assignable to type 'Promise<string>'.
"
`;
